packages/react-components/react-table/stories/Table/ResizableControlled.stories.tsx

```diff
@@ -0,0 +1,290 @@
+      label: 'You shared this in a Teams chat',
+      icon: <PeopleRegular />,
+    },
+  },
+];
+
+export const ResizableControlled = () => {
+  const [columns, setColumns] = useState<TableColumnDefinition<Item>[]>(columnsDef);
+  const [columnSizingOptions, setColumnSizingOptions] = useState<TableColumnSizingOptions>({
+    file: {
+      idealWidth: 300,
+      minWidth: 190,
+    },
+    author: {
+      minWidth: 170,
+      defaultWidth: 250,
+    },
+    lastUpdate: {
+      minWidth: 220,
+    },
+  });
+
+  const removeColumn = (index: number) => {
+    setColumns([...columns.slice(0, index), ...columns.slice(index + 1)]);
+  };
+
+  const addColumn = () => {
+    const currentColumnIds = columns.map(({ columnId }) => columnId);
+    const missingColumnIndex = columnsDef.findIndex(({ columnId }) => !currentColumnIds.includes(columnId));
+    if (missingColumnIndex !== -1) {
+      const missingColumn = columnsDef[missingColumnIndex];
+      setColumns(state => [...state.slice(0, missingColumnIndex), missingColumn, ...state.slice(missingColumnIndex)]);
+    }
+  };
+
+  const onWidthChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newWidth = parseInt(e.target.value, 10);
+    if (Number.isNaN(newWidth)) {
+      return;
+    }
+    setColumnSizingOptions(state => ({
+      ...state,
+      file: {
+        ...state.file,
+        idealWidth: newWidth,
+      },
+    }));
+  };
+
+  const onMinWidthChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newMinWIdth = parseInt(e.target.value, 10);
+    if (Number.isNaN(newMinWIdth)) {
+      return;
+    }
+    setColumnSizingOptions(state => ({
+      ...state,
+      file: {
+        ...state.file,
+        minWidth: newMinWIdth,
+      },
+    }));
+  };
+
+  const onColumnResize = React.useCallback((columnId: TableColumnId, width: number) => {
+    setColumnSizingOptions(state => ({
+      ...state,
+      [columnId]: {
+        ...state[columnId],
+        idealWidth: width,
+      },
+    }));
+  }, []);
+
+  const {
+    getRows,
+    sort: { getSortDirection, toggleColumnSort, sort },
+    columnSizing_unstable: columnSizing,
+    tableRef,
+  } = useTableFeatures(
+    {
+      columns,
+      items,
+    },
+    [
+      useColumnSizing_unstable({ columnSizingOptions, onColumnResize }),
+      useTableSort({ defaultSortState: { sortColumn: 'file', sortDirection: 'ascending' } }),
+    ],
+  );
+
+  const headerSortProps = (columnId: TableColumnId) => ({
+    onClick: (e: React.MouseEvent) => {
+      toggleColumnSort(e, columnId);
+    },
+    sortDirection: getSortDirection(columnId),
+  });
+
+  const rows = sort(getRows());
+
+  return (
+    <>
+      <p>
+        First column width: <input type="text" onChange={onWidthChange} value={columnSizingOptions.file.idealWidth} />
+      </p>
+      <p>
+        First column minWidth:{' '}
+        <input type="text" onChange={onMinWidthChange} value={columnSizingOptions.file.minWidth} />
```

Comment:
Let's use the `Input` component in the library, native `<input>` was nice for a prototype, but let's keep the look and feel for out docsite